iothub/device/src/Transport/Amqp/AmqpTransportHandler.cs

```diff
@@ -616,23 +616,32 @@ private void TwinMessageListener(Twin twin, string correlationId, TwinCollection
 
         protected override void Dispose(bool disposing)
         {
-            lock (_lock)
+            try
             {
-                if (_disposed)
+                if (Logging.IsEnabled)
                 {
-                    return;
+                    Logging.Enter(this, $"{nameof(DefaultDelegatingHandler)}.Disposed={_disposed}; disposing={disposing}", $"{nameof(AmqpTransportHandler)}.{nameof(Dispose)}");
                 }
 
-                base.Dispose(disposing);
+                lock (_lock)
+                {
+                    if (!_disposed)
+                    {
+                        if (disposing)
+                        {
+                            _closed = true;
+                            AmqpUnitManager.GetInstance()?.RemoveAmqpUnit(_amqpUnit);
+                        }
+                    }
 
+                    // the _disposed flag is inherited from the base class DefaultDelegatingHandler and is finally set to null there.
+                }
+            }
+            finally
+            {
                 if (Logging.IsEnabled)
-                    Logging.Info(this, nameof(disposing));
-
-                if (disposing)
                 {
-                    _closed = true;
-                    AmqpUnitManager.GetInstance()?.RemoveAmqpUnit(_amqpUnit);
-                    _disposed = true;
```

Comment:
Removed setting `_disposed` to `true`. The `_disposed` flag is inherited from the base class `DefaultDelegatingHandler` and is finally set to null there.

Reply:
I've reverted this change for now. I'll rework the inheritance disposal pattern of our pipeline (if necessary) in the next PR. I'm running some tests to confirm the behavior.